### PR TITLE
SDIT-939 Fix appointments create mapping retries

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/appointments/AppointmentsToNomisIntTest.kt [hunk 1]
```diff
@@ -375,7 +378,9 @@ class AppointmentsToNomisIntTest : SqsIntegrationTestBase() {
           await untilAsserted {
             mappingServer.verify(
               2,
-              postRequestedFor(urlEqualTo("/mapping/appointments")),
+              postRequestedFor(urlEqualTo("/mapping/appointments"))
+                .withRequestBody(matchingJsonPath("appointmentInstanceId", equalTo("$APPOINTMENT_INSTANCE_ID")))
+                .withRequestBody(matchingJsonPath("nomisEventId", equalTo("$EVENT_ID"))),
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/appointments/AppointmentsToNomisIntTest.kt:383]: This test should have caught the bug but it wasn't checking the JSON to make sure the right details are being passed in the retry message.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/appointments/AppointmentsToNomisIntTest.kt [hunk 2]
```diff
@@ -356,6 +356,9 @@ class AppointmentsToNomisIntTest : SqsIntegrationTestBase() {
           appointmentsApi.stubGetAppointmentInstance(id = APPOINTMENT_INSTANCE_ID, response = appointmentResponse)
 
           publishAppointmentEvent("appointments.appointment-instance.created")
+
+          await untilCallTo { appointmentsApi.getCountFor("/appointment-instances/$APPOINTMENT_INSTANCE_ID") } matches { it == 1 }
+          await untilCallTo { nomisApi.postCountFor("/appointments") } matches { it == 1 }
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/appointments/AppointmentsToNomisIntTest.kt:361]: This is needed to make the test work - something weird happens with Wiremock when trying to verify too early and you can't assert on the JSON.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/DomainEventListener.kt [hunk 3]
```diff
@@ -41,7 +43,15 @@ abstract class DomainEventListener(
...
-        RETRY_CREATE_MAPPING -> service.retryCreateMapping(sqsMessage.Message)
+        RETRY_CREATE_MAPPING -> runCatching { service.retryCreateMapping(sqsMessage.Message) }
+          .onFailure {
+            telemetryClient.trackEvent(
+              "create-mapping-retry-failure",
+              mapOf("retryMessage" to sqsMessage.Message),
+              null,
+            )
+            throw it
+          }
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/DomainEventListener.kt:54]: This is to make it easier to find the mapping create retry failures - previously there was no telemetry which is confusing as there is telemetry for everything else that happens (or fails) in this service.

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**